### PR TITLE
Fix commits to db not being executed on edit/delete/add ledger actions.

### LINKED ISSUE #2539: Add/edit/remove ledger action does not get commited to the DB until next commit
## Problem Definition

It's pretty amazing nobody noticed this so far, especially considering users do utilize ledger actions a lot. So both [add](https://github.com/rotki/rotki/blob/b7883aac6070c7cece4ec8ae1b854af24e3cebfd/rotkehlchen/db/ledger_actions.py#L133), [remove](https://github.com/rotki/rotki/blob/b7883aac6070c7cece4ec8ae1b854af24e3cebfd/rotkehlchen/db/ledger_actions.py#L151) and [editing](https://github.com/rotki/rotki/blob/b7883aac6070c7cece4ec8ae1b854af24e3cebfd/rotkehlchen/db/ledger_actions.py#L178) a ledger action does not commit to the DB.

At next commit they are included but ofcourse that can lead to serious problems if you don't commit to the DB soon after or close the application.

## Task

- [x] Write a test for these. Append db checks to the existing checks.
- [x] Fix it. Super simple. Just add a commit to the db in the end of the these functions.

### PR DESCRIPTION
**Pull request by yabirgb**
Also added tests to check that the DB has been updated after changes using this methods.

Closes #2539

### CODE REVIEW

**Comment by yabirgb** [comment]
Seems that the test that failed was already failing in the `bugfixes` branch

File: rotkehlchen/tests/unit/test_ledger_actions.py [hunk 1]
```diff
@@ -26,8 +26,13 @@ def test_serialize_deserialize_for_db():
 
 def test_all_action_types_writtable_in_db(database, function_scope_messages_aggregator):
     db = DBLedgerActions(database, function_scope_messages_aggregator)
+
+    # Query to check that the changes have been committed to the db
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_ledger_actions.py:30]: nitpick: This comment is not really needed. It's obvious. Remove it from here and rest of the tests

File: rotkehlchen/tests/unit/test_ledger_actions.py [hunk 2]
```diff
@@ -36,9 +41,78 @@ def test_all_action_types_writtable_in_db(database, function_scope_messages_aggr
...
+
+    # Data for the new entry
+    new_entry = LedgerAction(
+        identifier=identifier,
+        timestamp=2,
+        action_type=LedgerActionType.GIFT,
+        location=Location.EXTERNAL,
+        amount=FVal(3),
+        asset=A_ETH,
+        link="",
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_ledger_actions.py:102]: mind the styling. We use `' '` and not `" "`

**Review by LefterisJP — COMMENTED** [review]
Thanks @yabirgb 

Looks good. I have 2 nitpick comments that can be fixed now or later. But since you are missing a changelog entry better fix them now.

(nitpick comments are comments that don't stop a PR)

The thing you must do is add a changelog entry for the bug you fixed. I think you are familiar with how to do since you have already edited changelog.rst.

One other comment. In the issue I meant you can just add some DB checks in the api tests: https://github.com/rotki/rotki/blob/4fd6c0d734d38223f6cc4cf5e0d145da263e3208/rotkehlchen/tests/api/test_ledger_actions.py since they already test add/edit/delete there.

But what you did is also totally fine and fast also since it's unit tests.

File: rotkehlchen/tests/unit/test_ledger_actions.py (hunk 2 above)
Comment (yabirgb) [reply rotkehlchen/tests/unit/test_ledger_actions.py:102]: Ok will change this. I believe I copied it from an example in other test so might be in other places

File: docs/changelog.rst [hunk 3]
```diff
@@ -5,6 +5,7 @@ Changelog
 * :feature:`2507` Users can now delete imported trades and deposit/withdrawals from crypto.com via the purge data UI.
 * :bug:`2530` Poloniex should no longer display phantom LEND balances in rotki.
 * :bug:`2534` Aave v2 tokens not in Aave v1 should no longer have their balance double counted.
+* :bug:`2539` Now the effects of add/edit/remove operations related to ledger actions are committed to the database.
```
Comment (LefterisJP) [inline docs/changelog.rst:8]: This entry is correct as far as developing is concerned. But for our changelog as you can see from other entries we try to follow a more user-centric wording. So here I would suggest the following:

```suggestion
* :bug:`2539` The effects of adding/editing/removing a ledger actions will no longer be lost if rotki restarts right after the operation.
```

The idea is the user does not understand anything about DBs and stuff. But understands the bug being, to add something and then restart rotki, and then not see the added action

Comment (yabirgb) [reply docs/changelog.rst:8]: I see. Will take this into account for future modifications on the changelog

**Review by LefterisJP — APPROVED** [review]

**Pull request merged**